### PR TITLE
Add avg_cost and ppl for Transformer

### PR DESCRIPTION
**Pull request by guoshengCS**
Add avg_cost and ppl for Transformer.

### CODE REVIEW

File: fluid/neural_machine_translation/transformer/config.py [hunk 1]
```diff
@@ -15,6 +15,9 @@ class TrainTaskConfig(object):
     # the parameters for learning rate scheduling.
     warmup_steps = 4000
 
+    # the flag indicating to use average loss or sum loss when training.
+    use_avg = False
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/config.py]: use_avg --> use_avg_cost

File: fluid/neural_machine_translation/transformer/train.py [hunk 2]
```diff
@@ -171,14 +174,20 @@ def test(exe):
...
+                  (pass_id, batch_id, sum_cost_val, avg_cost_val,
+                   np.exp([min(avg_cost_val[0], 100)])))
         # Validate and save the model for inference.
-        val_cost = test(exe)
-        print("pass_id = " + str(pass_id) + " val_cost = " + str(val_cost))
+        val_sum_cost, val_avg_cost = test(exe)
+        pass_end_time = time.time()
+        time_consumed = pass_end_time - pass_start_time
+        print(
+            "epoch: %d, val sum loss: %f, val avg loss: %f, val ppl: %f, consumed %fs"
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/train.py]: exceed 80 columns.

File: fluid/neural_machine_translation/transformer/train.py [hunk 3]
```diff
@@ -114,7 +115,7 @@ def main():
         beta1=TrainTaskConfig.beta1,
         beta2=TrainTaskConfig.beta2,
         epsilon=TrainTaskConfig.eps)
-    optimizer.minimize(cost)
+    optimizer.minimize(avg_cost if TrainTaskConfig.use_avg else sum_cost)
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/train.py]: use_avg --> use_avg_cost

File: fluid/neural_machine_translation/transformer/config.py (hunk 1 above)
Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/config.py]: Done.

File: fluid/neural_machine_translation/transformer/train.py (hunk 3 above)
Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/train.py]: Done.

File: fluid/neural_machine_translation/transformer/train.py (hunk 2 above)
Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/train.py]: Done.

File: fluid/neural_machine_translation/transformer/train.py [hunk 4]
```diff
@@ -171,14 +174,20 @@ def test(exe):
...
             outs = exe.run(fluid.framework.default_main_program(),
                            feed=data_input,
-                           fetch_list=[cost],
+                           fetch_list=[sum_cost, avg_cost],
                            use_program_cache=True)
-            cost_val = np.array(outs[0])
-            print("pass_id = " + str(pass_id) + " batch = " + str(batch_id) +
-                  " cost = " + str(cost_val))
+            sum_cost_val, avg_cost_val = np.array(outs[0]), np.array(outs[1])
+            print("epoch: %d, batch: %d,  sum loss: %f, avg loss: %f, ppl: %f" %
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/train.py]: `  sum` --> ` sum`

Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/train.py]: Done.

**Review by pkuyym — APPROVED** [review]
LGTM

**Pull request merged**